---
title: "ATAC-seq footprint profiles using GGR Duke project A549 cell line data"
author: "Kaixuan Luo"
date: "7/31/2018"
output: 
  workflowr::wflow_html:
        toc: TRUE
        code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## functions
```{r}

##### Functions #####
## load and combine count matrices
load_combine_counts <- function(bam_basename, pwm_name, dir_count_matrix){
  cat("Loading count matrices ... \n")
  counts_fwd.df <- read.table(paste0(dir_count_matrix, "/", pwm_name, "/", pwm_name, "_", bam_basename, "_fwdcounts.m.gz"))
  counts_rev.df <- read.table(paste0(dir_count_matrix, "/", pwm_name, "/", pwm_name, "_", bam_basename, "_revcounts.m.gz"))
  
  ## the first 5 columns from "bwtool extract" are chr, start, end, name, and the number of data points
  counts_fwd.df <- counts_fwd.df[, -c(1:5)]
  counts_rev.df <- counts_rev.df[, -c(1:5)]
  
  colnames(counts_fwd.df) <- paste0("fwd", 1:ncol(counts_fwd.df))
  colnames(counts_rev.df) <- paste0("rev", 1:ncol(counts_rev.df))
  
  counts_combined.m <- as.matrix(cbind(counts_fwd.df, counts_rev.df))
  
  return(counts_combined.m)
}

## select candidate sites by mapability and PWM score cutoffs
select_sites <- function(sites.df, thresh_mapability=NULL, thresh_PWMscore=NULL, readstats_name=NULL){
  #  cat("loading sites ...\n")
  cat("Select candidate sites ... \n")
  
  if(!is.null(thresh_mapability)){
    cat("Select candidate sites with mapability >=", thresh_mapability, "\n")
    idx_mapability <- (sites.df[,"mapability"] >= thresh_mapability)
  }else{
    idx_mapability <- rep(TRUE, nrow(sites.df))
  }
  
  if(!is.null(thresh_PWMscore)){
    cat("Select candidate sites with PWM score >=", thresh_PWMscore, "\n")
    idx_pwm <- (sites.df[,"pwm_score"] >= thresh_PWMscore)
  }else{
    idx_pwm <- rep(TRUE, nrow(sites.df))
  }
  
  if(!is.null(readstats_name)){
    readstats.df <- read.table(readstats_name, header = F)
    ## if the readstats.df contains chrY, then it means the cell type is male, then the candidate sites should contain chrY,
    ## otherwise, the cell type is female, then the candidate sites on chrY should be removed.
    if( "chrY" %in% readstats.df[,1] ){
      cat("include chrY sites \n")
      idx_chr <- (sites.df[,1] != "")
    }else{
      cat("chrY NOT in the bam file, filter out chrY sites \n")
      ## remove chrY from candidate (motif) sites
      idx_chr <- (sites.df[,1] != "chrY")
    }
    
  }else{
    idx_chr <- rep(TRUE, nrow(sites.df))
  }
  
  idx_select <- which(idx_mapability & idx_pwm & idx_chr)
  
  return(idx_select)
}


## plot footprint profiles for selected sites
plot_footprint_sites <- function(tf_name, pwm_name, flank = 100, thresh_mapability = NULL, thresh_PWMscore = 0, num_top_sites = 1000){
  
  filename_sites <- paste0(dir_sites, "/", pwm_name, "_flank", flank, "_fimo_sites.bed")
  
  sites.df <- read.table(filename_sites, header = T, comment.char = "!", stringsAsFactors = F)
  
  colnames(sites.df) = c("chr", "start", "end", "name", "pwm_score", "strand", "p_value")

  readstats_name <-  paste0(dir_bam_idxstats, "/", "ATAC-seq_A549_rep1.idxstats.txt")
  
  idx_select <- select_sites(sites.df, thresh_mapability, thresh_PWMscore, readstats_name)
  
  sites.df <- sites.df[idx_select, c("chr", "start", "end", "name", "pwm_score", "strand", "p_value")]
  cat("Number of sites:", nrow(sites.df), "\n")
  
  counts_rep1.m <- load_combine_counts("ATAC-seq_A549_rep1", pwm_name, dir_count_matrix)[idx_select,]
  counts_rep2.m <- load_combine_counts("ATAC-seq_A549_rep2", pwm_name, dir_count_matrix)[idx_select,]
  counts_rep3.m <- load_combine_counts("ATAC-seq_A549_rep3", pwm_name, dir_count_matrix)[idx_select,]
  counts_combined.m <- counts_rep1.m + counts_rep2.m + counts_rep3.m
  
  ## Clip extreme values
  counts_combined.m[counts_combined.m > max_cuts] <- max_cuts

  cat("Dimension of", dim(counts_combined.m), "\n")
  
  if(nrow(counts_combined.m) != nrow(sites.df)){
    stop("Sites not matched!")
  }
  
  ## plot ATAC-seq cut profiles of most accessible sites 
  order_selected <- order(rowSums(counts_combined.m), decreasing = T)[1:num_top_sites]
  counts_selected.m <- counts_combined.m[order_selected,]
  counts_profile <- apply(counts_selected.m, 2, mean)

  par(mfrow = c(1,2))
  counts <- counts_profile[1:(length(counts_profile)/2)]
  plot(counts, type = "l", col = "blue", xlab = "Relative position (bp)", ylab = "Average counts", 
       main = "", xaxt = "n")
  mtext(text = paste(tf_name, "forward strand"), side = 3, line = 1, cex = 1)
  axis(1,at=c(1, flank+1, length(counts)-flank, length(counts)), labels=c(-flank, '','' ,flank), 
       cex.axis = 1, tck=-0.03, tick = T, cex = 1)
  
  counts <- counts_profile[(length(counts_profile)/2+1): length(counts_profile)]
  
  plot(counts, type = "l", col = "red", xlab = "Relative position (bp)", ylab = "Average counts", 
       main = "", xaxt = "n")
  mtext(text = paste(tf_name, "reverse strand"), side = 3, line = 1, cex = 1)
  axis(1,at=c(1, flank+1, length(counts)-flank, length(counts)), labels=c(-flank, '','' ,flank), 
       cex.axis = 1, tck=-0.03, tick = T, cex = 1)
  
  ## save counts matrix
  saveRDS(counts_selected.m, paste0(dir_matrix_examples, "/", pwm_name, "_", "ATAC-seq_A549_reps_counts_selected_sites.rds"))
  
  
}

```

## parameters
```{r}
ver_genome <- "hg38"
flank <- 100
thresh_mapability <- NULL
thresh_PWMscore <- 10
num_top_sites <- 1000
max_cuts <- 20 # Clip extreme values

bam_name_list <- c("ATAC-seq_A549_rep1.bam", "ATAC-seq_A549_rep2.bam", "ATAC-seq_A549_rep3.bam")

dir_data <- "~/Dropbox/research/ATAC_DNase/"

dir_count_matrix <- paste0(dir_data, "/ATAC-seq_GGR_Duke/results/ATAC-seq_count_matrix/")
dir_bam_idxstats <- paste0(dir_data, "/ATAC-seq_GGR_Duke/ATAC-seq_BAMfiles/")
dir_sites <- paste0(dir_data, "/motif_sites_JASPAR2018/hg38/candidate_sites/1e-4/")

dir_matrix_examples <- paste0(dir_data, "/ATAC-seq_GGR_Duke/results/ATAC-seq_count_matrix_examples/")
dir.create(dir_matrix_examples, showWarnings = F, recursive = T)

```

## CTCF
```{r}
tf_name <- "CTCF"
pwm_name <- "CTCF_MA0139.1_1e-4"

plot_footprint_sites(tf_name, pwm_name, flank, thresh_mapability, thresh_PWMscore, num_top_sites)

```

## REST
```{r}
tf_name <- "REST"
pwm_name <- "REST_MA0138.2_1e-4"

plot_footprint_sites(tf_name, pwm_name, flank, thresh_mapability, thresh_PWMscore, num_top_sites)

```

## HIF1A
```{r}
tf_name <- "HIF1A"
pwm_name <- "HIF1A_MA1106.1_1e-4"

plot_footprint_sites(tf_name, pwm_name, flank, thresh_mapability, thresh_PWMscore, num_top_sites)

```
